Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputMapper.cpp

```diff
@@ -408,32 +439,76 @@ namespace AzToolsFramework
 
     void QtEventToAzInputMapper::HandleKeyEvent(QKeyEvent* keyEvent)
     {
-        // Ignore key repeat events, they're unrelated to actual physical button presses.
+        const Qt::Key key = static_cast<Qt::Key>(keyEvent->key());
+        const QEvent::Type eventType = keyEvent->type();
+
+        // special handling for text events in edit mode
+        {
+            static Qt::Key lastKey = key;
+            static bool consumeAsText = false;
```

Comment:
Does this need to be static?

Reply:
+1

Reply:
As it turns out, the logic here is being overly greedy with consuming key events in their translation to text events.  Neither of these statics are necessary and will be removed.